hourly comparison interface

TrystanLea opened the pull request.

MrTimbones commented on www/views/compare.html:
```diff
@@ -0,0 +1,200 @@
+    <a href="." title="Home"><i aria-hidden="true" class="fas fa-home"></i></a>
+    <a href="stats.html" title="30 Day Stats"><i aria-hidden="true" class="fas fa-table"></i></a>
+    <a href="costs.html" title="Running Costs"><i aria-hidden="true" class="fas fa-coins"></i></a>
+    <a href="graph.html" title="Comparison Charts" class="active"><i aria-hidden="true" class="fas fa-chart-line"></i></a>
```
We're going to have to split this into two links, and think of better titles, or we have all the chart features on one page.

The hardest part is picking an appropriate icon. I'm picking from [fontawesome v5.15](https://fontawesome.com/v5/search?o=r&s=solid), and best I can find is `<i class="fas fa-object-group"></i>`. v6 has fa-chart-scatter, but I couldn't get it to work.

TrystanLea replied:
ok sounds good, I think getting all the chart features on the same page may be too much of a challenge for now. 
I think there will be another graph page shortly to provide the half hourly profile for all systems...

MrTimbones commented on www/views/compare.html:
```diff
@@ -0,0 +1,200 @@
+</html>
+
+
+<script>
```
Should we move this script into its own file?

TrystanLea replied:
sure will do that

MrTimbones commented:
Can we make the 2nd system selector default to `None` so that people can view a single system if they wish, with the 2nd system being optional.

**Future idea**: allow selection of the same system on both sides, but with different dates.

TrystanLea commented on scrape_kwh.php:
```diff
@@ -20,6 +20,7 @@
   # read each row and append kWh readings for last year
   while (($row = fgetcsv($input, 1000, "\t")) !== FALSE) {
     $system = ['id' => count($data) + 1];
+    echo json_encode($system)."\n";
```
they are not essential, Id like to cut down the number of requests made here, I think we could get more from the stats api.. will look into it.

MrTimbones replied:
Are you worried about load on the server as the number of systems grows? This script is only really run once-a-day, around midnight, and we can put in some sleeps to spread the load if that will help.

TrystanLea commented:
Hello @MrTimbones 

I've moved the header, navigation and footer for each page to a theme file and moved the javascript out into js files.
Let me know if you are happy for me to merge this.

Cheers!

MrTimbones commented:
I can't get this branch to run on my test server, but I guess it works fine for you?

TrystanLea commented:
yes works fine here, just requires modrewrite enabled and php (same requirements as for emoncms). Do you see any errors? any clues as to why it's not working?

MrTimbones commented:
I'll figure it out. If it works for you and you're confident that it'll work on the main server, I'm happy for this to be merged,

TrystanLea commented:
ok great, thanks!

TrystanLea commented:
Ok turns out I had missed pushing up index.php! as it was in .gitignore :) fixed now
